bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleRegistryImpl.java

```diff
@@ -376,11 +377,8 @@ private Rule resolveRuleByTemplate(Rule rule) {
      */
     private void updateRuleTemplateMapping(String templateUID, String ruleUID, boolean resolved) {
         synchronized (this) {
-            Set<String> ruleUIDs = mapTemplateToRules.get(templateUID);
-            if (ruleUIDs == null) {
-                ruleUIDs = new HashSet<>();
-                mapTemplateToRules.put(templateUID, ruleUIDs);
-            }
+            Set<String> ruleUIDs = Objects
+                    .requireNonNull(mapTemplateToRules.computeIfAbsent(templateUID, k -> new HashSet<>()));
             if (resolved) {
```

Comment:
I'm sorry, but I completely fail to see how this is simplifying the code. I doubt there are any performance benefits, and the old code is certainly much clearer as to what actually happens. Especially, `Objects.requireNonNull()` obscures the whole thing, and although cheap, it still is a completely useless operation that does require pushing and popping the stack etc.

I think a better description would be "Trade readability for slightly shorter code".